src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/beanvalidation/BeanValidationDiagnosticsCollector.java

```diff
@@ -160,26 +161,29 @@ private void validAnnotation(PsiElement element, PsiAnnotation annotation, Strin
                             && !type.equals(PsiType.LONG)
                             && !type.equals(PsiType.FLOAT)
                             && !type.equals(PsiType.DOUBLE)) {
-                        diagnostics.add(createDiagnostic(element, (PsiJavaFile) element.getContainingFile(), "The @"
-                                + getSimpleName(annotationName) + " annotation can only be used on \n- BigDecimal \n- BigInteger"
-                                + "\n- byte, short, int, long, float, double (and their respective wrappers) \n type "
-                                + source, DIAGNOSTIC_CODE_INVALID_TYPE, annotationName, DiagnosticSeverity.Error));
+                        String source = isMethod ?
+                                Messages.getMessage("AnnotationPositiveMethods", "@" + getSimpleName(annotationName)) :
+                                Messages.getMessage("AnnotationPositiveFields", "@" + getSimpleName(annotationName));
+                        diagnostics.add(createDiagnostic(element, (PsiJavaFile) element.getContainingFile(),
+                                source, DIAGNOSTIC_CODE_INVALID_TYPE, annotationName, DiagnosticSeverity.Error));
                     }
                 } else if (matchedAnnotation.equals(NOT_BLANK)) {
                     if (!type.getCanonicalText().endsWith(STRING)
                             && !type.getCanonicalText().endsWith(CHAR_SEQUENCE)) {
+                        String source = isMethod ?
+                                Messages.getMessage("AnnotationStringMethods", "@" + getSimpleName(annotationName)) :
+                                Messages.getMessage("AnnotationStringFields", "@" + getSimpleName(annotationName));
                         diagnostics.add(createDiagnostic(element, (PsiJavaFile) element.getContainingFile(),
-                                "The @" + getSimpleName(annotationName)
-                                        + " annotation can only be used on String and CharSequence type " + source,
-                                DIAGNOSTIC_CODE_INVALID_TYPE, annotationName, DiagnosticSeverity.Error));
+                                source, DIAGNOSTIC_CODE_INVALID_TYPE, annotationName, DiagnosticSeverity.Error));
                     }
                 } else if (matchedAnnotation.equals(PATTERN)) {
```

Comment:
Possibly, but from a design point of view the NotBlank and the Pattern annotations are not closely related, they just have similar error handling. The way this came about is probably they were written at different times. Could perhaps pull the code into a method but I like the idea that the individual annotations have their own `else if` rather than combining them. Want to do that? Keep in mind, with dual maintenance, we would need to update lsp4jakarta also.

Reply:
That sounds good - a method call while keeping the `else if` blocks independent. I'm just now wondering if changes like this should be made during the refactoring work. Up to you if you would like to make this change now or later.

Reply:
Ok